daemon/src/model/cfd.rs

```diff
@@ -1352,13 +1354,61 @@ pub fn calculate_profit_at_price(
 
 #[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
 pub struct Cet {
-    pub tx: Transaction,
+    #[serde(with = "::bdk::bitcoin::util::amount::serde::as_sat")]
+    pub maker_amount: Amount,
+    #[serde(with = "::bdk::bitcoin::util::amount::serde::as_sat")]
+    pub taker_amount: Amount,
     pub adaptor_sig: EcdsaAdaptorSignature,
 
     // TODO: Range + number of digits (usize) could be represented as Digits similar to what we do
     // in the protocol lib
     pub range: RangeInclusive<u64>,
     pub n_bits: usize,
+
+    pub txid: Txid,
+}
+
+impl Cet {
+    /// Build an actual `Transaction` out of the payout information
+    /// stored in `Self`, together with the input and the output
+    /// addresses.
+    ///
+    /// The order of the outputs matters.
+    ///
+    /// We verify that the TXID of the resulting transaction matches
+    /// the TXID with which `Self` was constructed.
+    pub fn to_tx(
+        &self,
+        (commit_tx, commit_descriptor): (&Transaction, &Descriptor<PublicKey>),
+        maker_address: &Address,
+        taker_address: &Address,
+    ) -> Result<Transaction> {
+        let tx = Transaction {
+            version: 2,
+            input: vec![TxIn {
+                previous_output: commit_tx.outpoint(&commit_descriptor.script_pubkey())?,
```

Comment:
I am curious whether or not we can make this function infallible if we pass the outpoint in and make all fields private. Assuming you construct the struct correctly, the output of this function would be what you want. That could shave off the storage of the txid.

Let's explore this at some point.

Reply:
I would very much like to not store the `Txid`, which is kind of redundant info. Since you seem to have a good idea, maybe record this somewhere :D